tsfc/coffee.py

```diff
@@ -296,7 +299,7 @@ def _expression_scalar(expr, parameters):
     if isnan(expr.value):
         return coffee.Symbol("NAN")
     else:
-        return coffee.Symbol(("%%.%dg" % (PRECISION - 1)) % expr.value)
+        return coffee.Symbol(("%%.%dg" % (parameters.precision - 1)) % expr.value)
```

Comment:
You mean same output of tsfc and quad/uflacs? Yes. COFFEE generated arrays seem to have correct number of significant digits. Sometimes they're pone digit shorter but that could be trailing zero. Minus sign or number of digits before comma do not seem to play a role.

`return coffee.Symbol(("%%.%dg" % (parameters.precision - 1)) % expr.value)` yields one less digit than expected.